src/Symfony/Component/Notifier/Bridge/Matrix/MatrixTransport.php

```
@@ -0,0 +1,226 @@
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Frank Schulze <frank@akiber.de>
+ */
+final class MatrixTransport extends AbstractTransport
+{
+    public function __construct(
+        #[\SensitiveParameter] private string $accessToken,
+        private bool $ssl,
+        ?HttpClientInterface $client = null,
+        ?EventDispatcherInterface $dispatcher = null,
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return \sprintf('matrix://%s', $this->getEndpoint(false));
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof MatrixOptions);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        if (null === $message->getRecipientId()) {
+            throw new LogicException('Recipient id is required.');
+        }
+
+        $recipient = match (mb_substr($message->getRecipientId(), 0, 1)) {
+            '@' => $this->getDirectMessageChannel($message->getRecipientId()),
+            '!' => $message->getRecipientId(),
+            '#' => $this->getRoomFromAlias($message->getRecipientId()),
+            default => throw new UnsupportedRecipientTypeException(__CLASS__, mb_substr($message->getRecipientId(), 0, 1)),
+        };
+
+        return $this->sendMessage($recipient, $message);
+    }
+
+    protected function sendMessage(
+        string $recipientId,
+        MessageInterface $message,
+    ): SentMessage {
+        /** @var MatrixOptions $options */
+        $options = $message->getOptions();
```

this should throw an `UnsupportedOptionsException` if `$options` is not `null` but no `MatrixOptions` instance

---

thats what `supports()` already do and to do it in there will be consistent to other bridges see https://github.com/symfony/symfony/pull/59377/files#r1908353006